RNMobile: Add integration test guide

guarani opened the pull request:
## Description

Adds a guide to integration tests on Gutenberg Mobile.

## How has this been tested?
Read through the guide to ensure legibility and accuracy.

## Types of changes
- Updated docs

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [x] I've tested my changes with keyboard and screen readers. <!-- Instructions: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/accessibility-testing.md -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/code/native-mobile.md -->

Commit:
Add integration test guide

Co-authored-by: Carlos Garcia <fluiddot@gmail.com>

fluiddot commented on docs/contributors/code/native-mobile-integration-test-guide.md:
```diff
@@ -0,0 +1,368 @@
+-   Expect correct element behaviour
+-   Cleanup
+
+## Setup
```
I'd use H3 (`###`) for the sections under `Anatomy of an integration test`, wdyt?

fluiddot replied:
I see that the following main sections like "Common flows" use H1 (`#`) so maybe for "Anatomy of an integration test" we could do the same and keep these sections with H2 (`##`).

guarani replied:
You're right that the section titles and subtitles have inconsistent sizes. I'm going to tweak it a little, because in my opinion only the document title itself should be H1 (`#`).

guarani replied:
I've made changes to only use H1 (`#`) in the document title and have changed the sizes of the other smaller headings. I've also made the "Anatomy of an integration test" a table of contents, because that seems to fit well with the structure of the doc.

fluiddot commented on docs/contributors/code/native-mobile-integration-test-guide.md:
```diff
@@ -0,0 +1,368 @@
+
+## Expect correct element behaviour
+
+After querying elements and firing events, we have to verify that the logic works as expected, for this purpose we can use the same expect function from Jest as we use in unit tests.
```
The "expect" word has inline code format in the original document.
```suggestion
After querying elements and firing events, we have to verify that the logic works as expected, for this purpose we can use the same `expect` function from Jest as we use in unit tests.
```

guarani replied:
Thanks for catching and fixing that 🙇

fluiddot reviewed (commented):
I reviewed the document and it looks great to me 🎊, I only added a couple of comments that are minor, thanks @guarani for adding the guide 🙇 !

Commit:
Update docs/contributors/code/native-mobile-integration-test-guide.md

Co-authored-by: Carlos Garcia <fluiddot@gmail.com>

guarani commented on docs/contributors/code/native-mobile-integration-test-guide.md:
```diff
@@ -0,0 +1,368 @@
+# Integration Test Guide
```
```suggestion
# React Native Integration Test Guide
```

guarani replied:
I'm updating the title here to coincide with the title we gave it [here](https://github.com/WordPress/gutenberg/pull/33833/files#diff-4277f4c93ff85051e589fc245170b43b437cf5c004904bfd16b31287f4d35151R27).

Commit:
Update docs/contributors/code/native-mobile-integration-test-guide.md

Commit:
Tweaked heading sizes and added table of contents

Commit:
Make screenshot wider for better legibility

guarani commented:
@fluiddot, I've changed the document title https://github.com/WordPress/gutenberg/pull/33833/commits/01b9e875d7e7a8e4e7e461f28427135eae2d3745, change the heading sizes and added a table of contents https://github.com/WordPress/gutenberg/pull/33833/commits/6d90edda95c62076e8eabce12ff50f1c511bc541, and made the screenshot of the Accessibility Inspector a little wider for better readability 99179203bf5eb16b2f71feadd3fe184a5d147504.
It's now ready for a re-review.

fluiddot reviewed (approved):
LGTM 🎊 !

> @fluiddot, I've changed the document title 01b9e87, change the heading sizes and added a table of contents 6d90edd, and made the screenshot of the Accessibility Inspector a little wider for better readability 9917920.
It's now ready for a re-review.

Thanks for the changes, I really liked the idea of the table of contents 💯 .

guarani commented:
Thanks for all the work that went into the documentation! 👍
